docs/dom-testing-library/api-async.mdx

```diff
@@ -62,6 +64,11 @@ options.
 This can be useful if you have a unit test that mocks API calls and you need to
 wait for your mock promises to all resolve.
 
+The callback passed to `waitFor` _must throw an error_ for as long as it should
+be waited for. The error must be of the same form as failed `expect` errors.
+The simplest way to achieve that is to write the condition as an expectation
+(see example above).
+
```

Comment:
I like the change, but I think we should also revert this change and remove the paragraph.

```suggestion
```